### PR TITLE
Rename lexer dir to lex

### LINKED ISSUE #3070: Restructuring & namespacing plan for the toolchain
I wanted to capture a discussion that several of us had around a restructuring of the toolchain to give ourselves a bit of a better directory layout and nomenclature.

Overall, the plan is to move towards a structure of major components where there is a reasonable namespace we can use to have a regular name component of code within that component. That is, rather than `SemanticsNode`, we'll have something structurally more like `Semantics::Node`.

As part of that, we noted a couple of things:
- There are really two layers inside of the `semantics` directory currently: the semantics-IR, and the code to check a parse tree and build that IR from it.
- The names we're using aren't awesome, and that showed up when trying to give command line flags that reflect the various stages.

One of the challenges we have is that the command line wants verb-shaped names as they're about the _action_ of that phase. But forming nouns out of the verbs for the phases doesn't follow a consistent structure. For example, "parser" is great, but "lowerer" is really awkward. So our thought was to switch to naming things after the verb everywhere.

The outcome of this (if I remember correctly -- please correct if I've misremembered) was to move to the following rough name / structure.

- `lex` directory, and if needed a `Carbon::Lex` namespace
- `parse` directory, and if needed a `Carbon::Parse` namespace
- `check` directory for the semantic & type checking logic that diagnoses invalid Carbon code and produces a valid semantics IR, and `Carbon::Check` namespace
- `lower` directory and `Carbon::Lower` namespace
- `codegen` directory and `Carbon::Codegen` namespace, treating "codegen" as a word

We talked about what to name the component that provides the semanitics IR definition and APIs, but didn't come up with anything that I think folks were delighted by. Some of the candidates:
- `SIR` -- not great if pronounced "sir".
- `SemIR` -- maybe pronounced "simmer"?
- `SemaIR` -- the abbreviation "sema" in Clang hasn't been enjoyable, especially for non-native speakers
- `CarbonIR` -- unclear if this is an IR written *in* Carbon or *for* Carbon. Also, we might end up with more IRs if we want to do high level optimizations.

We didn't really talk about how we want to handle the components that aren't named after a phase of compilation. Personally, I think these are maybe fine to have subdirectories but no namespace as they will tend to be widely used. And I'm also OK with the current names we have there. But flagging it in case others disagree.

We're not planning to rush out and change everything here. This issue is to try to capture the plan for whenever seems like an opportune moment to minimize the churn for folks with pending PRs, and to  make sure everyone is happy with the outcome. It should also allow us to start opportunistically using this nomenclature where needed, for example in the command line.

### PR DESCRIPTION
**Pull request by jonmeow**
Continuing with #3070. Just a dir and file rename (only prefix change is lexer_file_test). Everything in the lex dir should be marked as a move.

Note, I think this closes #3070. There may still be further cleanup later, but the organizational changes suggested there are being completed.

### CODE REVIEW

File: toolchain/lex/lex_helpers.h [hunk 1]
```diff
@@ -2,8 +2,8 @@
 // Exceptions. See /LICENSE for license information.
```
Comment (zygoloid) [inline toolchain/lex/lex_helpers.h]: Would it make sense to rename this file to just "helpers.h"?

I'm happy for that to be done in a separate PR (if at all).

**Review by zygoloid — APPROVED** [review]

File: toolchain/lex/lex_helpers.h (hunk 1 above)
Comment (jonmeow) [reply toolchain/lex/lex_helpers.h]: Double-checked that "helpers.h" is still preferred with "test_helpers.h" here too. Renamed.

**Comment by chandlerc** [comment]
The prior commit is in the merge queue. Trying out adding this one to see if it DTRT and resolves the merge.

**Comment by chandlerc** [comment]
That would be "no, it won't".

Going to see if I can add it when the other commit lands.

**Comment by chandlerc** [comment]
Not auto-resolved. I'm going to try to resolve the conflicts and merge this so that all the renames are landed and done without more waiting and/or churn tomorrow.
